providers/openstack/kamaji/1-30/cluster-class/templates/kamaji-control-plane-template.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha1
+kind: KamajiControlPlaneTemplate
+metadata:
+  name: {{ .Release.Name }}-{{ .Chart.Version }}-control-plane
+spec:
+  template:
+    spec:
```

Comment:
The field `dataStoreName` is probably missing here, attaching cso logs
```
{"level":"ERROR","time":"2024-06-17T08:01:21.390Z","file":"kube/kube.go:107","message":"failed to apply object","controller":"clusterstackrelease","controllerGroup":"clusterstack.x-k8s.io","controllerKind":"ClusterStackRelease","ClusterStackRelease":{"name":"openstack-kamaji-1-30-v0-sha-11930ee","namespace":"default"},"namespace":"default","name":"openstack-kamaji-1-30-v0-sha-11930ee","reconcileID":"59c2788e-1bb8-4934-92cd-c297d26cc72c","obj":"controlplane.cluster.x-k8s.io/v1alpha1, Kind=KamajiControlPlaneTemplate","name":"openstack-kamaji-1-30-v0-sha.11930ee-control-plane","namespace":"","error":"failed to apply object: KamajiControlPlaneTemplate.controlplane.cluster.x-k8s.io \"openstack-kamaji-1-30-v0-sha.11930ee-control-plane\" is invalid: spec.template.spec.dataStoreName: Required value","stacktrace":"github.com/SovereignCloudStack/cluster-stack-operator/pkg/kube.(*kube).Apply\n\t/src/cluster-stack-operator/pkg/kube/kube.go:107\ngithub.com/SovereignCloudStack/cluster-stack-operator/internal/controller.(*ClusterStackReleaseReconciler).templateAndApply\n\t/src/cluster-stack-operator/internal/controller/clusterstackrelease_controller.go:286\ngithub.com/SovereignCloudStack/cluster-stack-operator/internal/controller.(*ClusterStackReleaseReconciler).Reconcile\n\t/src/cluster-stack-operator/internal/controller/clusterstackrelease_controller.go:169\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/src/cluster-stack-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/src/cluster-stack-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/src/cluster-stack-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/src/cluster-stack-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227"}
```

Reply:
Thank you, I will look into it!

Reply:
Please, try it with `control-plane-components.yaml` from https://github.com/clastix/cluster-api-control-plane-provider-kamaji/pull/111

Reply:
Cool, I will try and let you know

Reply:
After applying the new CRDs, the clusterstackrelease is now ready
```
$ k get clusterstackreleases.clusterstack.x-k8s.io 
NAME                                   K8S VERSION   READY   AGE     REASON   MESSAGE
openstack-kamaji-1-30-v0-sha-11930ee   v1.30.1       true    5h20m   
```